lib/Yield/Assembly/HTML.cpp

```diff
@@ -5,65 +5,43 @@
 // This file is distributed under the MIT License. See LICENSE.md for details.
 //
 
-#include "llvm/Support/FormatVariadic.h"
-
+#include "revng/ADT/Concepts.h"
 #include "revng/EarlyFunctionAnalysis/ControlFlowGraph.h"
 #include "revng/Model/Binary.h"
 #include "revng/Yield/ControlFlow/FallthroughDetection.h"
 #include "revng/Yield/Function.h"
 #include "revng/Yield/HTML.h"
+#include "revng/Yield/Support/PTML.h"
 
-namespace tags {
-
-static constexpr auto Function = "function";
-static constexpr auto LabeledBlock = "labeled-block";
-static constexpr auto BasicBlock = "basic-block";
-static constexpr auto FunctionLabel = "function-label";
-static constexpr auto BasicBlockLabel = "basic-block-label";
-
-static constexpr auto Instruction = "instruction";
-static constexpr auto InstructionAddress = "instruction-address";
-static constexpr auto InstructionBytes = "instruction-bytes";
-
-static constexpr auto InstructionMnemonic = "mnemonic";
-static constexpr auto InstructionMnemonicPrefix = "mnemonic-prefix";
-static constexpr auto InstructionMnemonicSuffix = "mnemonic-suffix";
-static constexpr auto InstructionOpcode = "instruction-opcode";
-
-static constexpr auto Comment = "comment";
-static constexpr auto Error = "error";
-static constexpr auto ImmediateValue = "immediate-value";
-static constexpr auto MemoryOperand = "memory-operand";
-static constexpr auto Register = "register";
-
-static constexpr auto CommentIndicator = "comment-indicator";
-static constexpr auto LabelIndicator = "label-indicator";
-
-static constexpr auto FunctionLink = "function-link";
-static constexpr auto BasicBlockLink = "basic-block-link";
-static constexpr auto InstructionLink = "instruction-link";
+using yield::ptml::str;
+using yield::ptml::Tag;
+namespace attributes = yield::ptml::attributes;
+namespace ptmlScopes = yield::ptml::scopes;
+namespace tags = yield::ptml::tags;
 
-static constexpr auto BasicBlockOwner = "basic-block-owner";
+namespace tokenTypes {
 
-static constexpr auto InstructionTarget = "instruction-target";
-static constexpr auto InstructionTargets = "instruction-targets";
+static constexpr auto Label = "asm.label";
+static constexpr auto LabelIndicator = "asm.label-indicator";
+static constexpr auto CommentIndicator = "asm.comment-indicator";
+static constexpr auto Mnemonic = "asm.mnemonic";
+static constexpr auto MnemonicPrefix = "asm.mnemonic-prefix";
+static constexpr auto MnemonicSuffix = "asm.mnemonic-suffix";
+static constexpr auto ImmediateValue = "asm.immediate-value";
+static constexpr auto MemoryOperand = "asm.memory-operand";
+static constexpr auto Register = "asm.register";
 
-static constexpr auto Whitespace = "whitespace";
-static constexpr auto Untagged = "untagged";
+} // namespace tokenTypes
 
-} // namespace tags
+namespace scopes {
 
-namespace templates {
+static constexpr auto Function = "asm.function";
+static constexpr auto BasicBlock = "asm.basic-block";
+static constexpr auto Instruction = "asm.instruction";
```

Comment:
I'd like to see all this stuff documented in the [PTML pad](https://pad.rev.ng/emEtVjiLTz2dRCnf9avSng#).

Reply:
Done